x/programs/examples/token.go

```diff
@@ -33,8 +34,9 @@ const (
 	Balance
 )
 
-func NewToken(log logging.Logger, engine *engine.Engine, programBytes []byte, db state.Mutable, cfg *runtime.Config, imports host.SupportedImports, maxUnits uint64) *Token {
+func NewToken(id ids.ID, log logging.Logger, engine *engine.Engine, programBytes []byte, db state.Mutable, cfg *runtime.Config, imports host.SupportedImports, maxUnits uint64) *Token {
 	return &Token{
+		id:           id,
```

Comment:
Not sure if this makes sense. 

But I also don't quite understand why there is so much Go code to define "Token" like this. This should all only be in Rust IMO